packages/http-client-java/emitter/src/common/client.ts

```diff
@@ -11,6 +11,11 @@ export interface Client extends Aspect, CrossLanguageDefinition {
   security: Security;
 
   serviceVersion?: ServiceVersion; // apiVersions is in
+
+  parent?: Client;
+  subClients: Array<Client>;
+  publicBuilder: boolean;
+  publicParentAccessor: boolean;
```

Comment:
What's the relation between these two parameters and [`subclient.initialization.access`/`client.<clientaccessor_method>.access`](https://github.com/Azure/typespec-azure/issues/1702)?

Reply:
Better naming the two parameters as `hasPublicBuilder` amd `hasPublicParentAccessor`

Reply:
To Xiaofei: Yes, it is intended to capture the access value from the 2 `access` from TCGC.

But since TCGC hasn't finalized the design and impl, we would just use our convention (and the emitter option) to decide the access mode.

To Haoling: Sure, will check the name again.